File: LovePlay-UIKit/LovePlay/Class/Zone/View/DiscussListCell.m
//
//  DiscussListCell.m
//  LovePlay
//
//  Created by Yuns on 2017/2/3.
//  Copyright © 2017年 Yuns. All rights reserved.
//

#import "DiscussListCell.h"
#import "DiscussListModel.h"

@interface DiscussListCell ()
//UI
@property (nonatomic, strong) UILabel *titleTextNode;
@property (nonatomic, strong) UILabel *descriptionTextNode;
@property (nonatomic, strong) UIView *underLineNode;
//Data
@property (nonatomic, strong) ForumThread *forumThread;
@end

@implementation DiscussListCell

+ (instancetype)cellWithTableView:(UITableView *)tableView
{
    static NSString *ID = @"DiscussListCell";
    DiscussListCell *cell = [tableView dequeueReusableCellWithIdentifier:ID];
    if (cell == nil) {
        cell = [[DiscussListCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:ID];
    }
    return cell;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        self.selectionStyle = UITableViewCellSelectionStyleNone;
        [self addSubNodes];
        [self mas_subViews];
    }
    return self;
}

- (void)addSubNodes
{
    [self.contentView addSubview:self.titleTextNode];

    [self.contentView addSubview:self.descriptionTextNode];

    [self.contentView addSubview:self.underLineNode];
}

- (void)mas_subViews
{
    [_titleTextNode mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self.contentView).offset(10);
        make.left.equalTo(self.contentView).offset(10);
        make.right.equalTo(self.contentView).offset(-10);
    }];
    
    [_descriptionTextNode mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(_titleTextNode.mas_bottom).offset(10);
        make.left.right.equalTo(_titleTextNode);
    }];
    
    [_underLineNode mas_makeConstraints:^(MASConstraintMaker *make) {
        make.height.mas_equalTo(0.5);
        make.top.equalTo(_descriptionTextNode.mas_bottom).offset(10);
        make.left.bottom.right.equalTo(self.contentView);
    }];
}

#pragma mark - public
- (void)setupForumThread:(ForumThread *)forumThread
{
    _forumThread = forumThread;
    _titleTextNode.text = forumThread.subject;
    _descriptionTextNode.text = [NSString stringWithFormat:@"%@回复   %@   %@",forumThread.replies, [forumThread.lastpost stringByReplacingOccurrencesOfString:@"&nbsp;" withString:@""], forumThread.lastposter];
}

#pragma mark - setter / getter
- (UILabel *)titleTextNode
{
    if (!_titleTextNode) {
        UILabel *titleTextNode = [[UILabel alloc] init];
        titleTextNode.numberOfLines = 2;
        titleTextNode.font = [UIFont systemFontOfSize:14];
        titleTextNode.textColor = RGB(36, 36, 36);
        _titleTextNode = titleTextNode;
    }
    return _titleTextNode;
}

- (UILabel *)descriptionTextNode
{
    if (!_descriptionTextNode) {
        UILabel *descriptionTextNode = [[UILabel alloc] init];
        descriptionTextNode.font = [UIFont systemFontOfSize:10];
        descriptionTextNode.textColor = RGB(150, 150, 150);
        _descriptionTextNode = descriptionTextNode;
    }
    return _descriptionTextNode;
}

- (UIView *)underLineNode
{
    if (!_underLineNode) {
        UIView *underLineNode = [[UIView alloc] init];
        underLineNode.backgroundColor = RGB(222, 222, 222);
        _underLineNode = underLineNode;
    }
    return _underLineNode;
}


@end

File: LovePlay-Swift3/LovePlay/Class/Zone/View/DiscussListCell.swift
//
//  DiscussListCell.swift
//  LovePlay
//
//  Created by weiying on 2017/6/6.
//  Copyright © 2017年 yuns. All rights reserved.
//

import UIKit

class DiscussListCell: UITableViewCell {

    class func cellWithTableView(tableView : UITableView) -> DiscussListCell {
        let ID = NSStringFromClass(self)
        var cell : DiscussListCell? = tableView.dequeueReusableCell(withIdentifier: ID) as? DiscussListCell
        if cell == nil {
            cell = DiscussListCell(style: .default, reuseIdentifier: ID)
        }
        return cell!
    }
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.selectionStyle = .none
        self.addSubViews()
        self.snp_subViews()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    // MARK: - private
    private func addSubViews() {
        self.contentView.addSubview(self.titleTextLabel)
        self.contentView.addSubview(self.descTextLabel)
        self.contentView.addSubview(self.underLineView)
    }
    
    private func snp_subViews() {
        self.titleTextLabel.snp.makeConstraints { (make) in
            make.top.equalToSuperview().offset(10)
            make.left.equalToSuperview().offset(10)
            make.right.equalToSuperview().offset(-10)
        }
        
        self.descTextLabel.snp.makeConstraints { (make) in
            make.top.equalTo(self.titleTextLabel.snp.bottom).offset(10)
            make.left.right.equalTo(self.titleTextLabel)
        }
        
        self.underLineView.snp.makeConstraints { (make) in
            make.height.equalTo(0.5)
            make.top.equalTo(self.descTextLabel.snp.bottom).offset(10)
            make.left.bottom.right.equalToSuperview()
        }
    }
    
    fileprivate var _listModel : DiscussListModel?
    var listModel : DiscussListModel? {
        set {
            _listModel = newValue
        }
        
        get {
            return _listModel
        }
    }
    
    //MARK: - setter / getter
    lazy var titleTextLabel : UILabel = {
        let label : UILabel = UILabel()
        label.numberOfLines = 2
        label.font = UIFont.systemFont(ofSize: 14)
        label.textColor = RGB(r: 36, g: 36, b: 36)
        return label
    }()
    
    lazy var descTextLabel : UILabel = {
        let label : UILabel = UILabel()
        label.font = UIFont.systemFont(ofSize: 10)
        label.textColor = RGB(r: 150, g: 150, b: 150)
        return label
    }()
    
    lazy var underLineView : UIView = {
        let lineView : UIView = UIView()
        lineView.backgroundColor = RGB(r: 222, g: 222, b: 222)
        return lineView
    }()
}
